ouroboros-consensus/test/infra-test/Test/Ouroboros/Consensus/ChainGenerator/Tests/BitVector.hs

```diff
@@ -141,3 +158,56 @@ prop_findIthZeroInV testSetup = case testSetup of
                       $ QC.counterexample "There are not testIndex-many post-polarization-0s preceding i!"
                       $ targetsInPrecedingWords QC.=== C.getCount testIndex
                   )
+
+
+data FillInWindowSetup =
+    FillInWindowSetup
+      SomePol
+      Int -- ^ k
+      Int -- ^ size of the window to fill
+      QCGen -- ^ random generator
+      (V.Vector S) -- ^ the vector to fill
+  deriving (Show)
```

Comment:
What about giving the fields names, such that eg the `Show` instance is more instructive (which is used when a QuickCheck test fails)?

Reply:
I wrote a `Show` instance by hand. I'm discovering I have a strong will to avoid polluting the namespace if I can.

Reply:
Makes sense :+1:, soon:tm: we will be able to drop 8.10 and can finally use `NoFieldSelectors` :pray: